5.drug-dependency/scripts/3.t-tests.py

```diff
@@ -0,0 +1,207 @@
+    disease_drug_df = drug_df_with_disease[drug_df_with_disease['OncotreePrimaryDisease'] == disease].drop(columns=['ModelID', 'OncotreePrimaryDisease'])
+
+    # Filter the drug matrix for the rest of the dataset (all other diseases)
+    other_drug_df = drug_df_with_disease[drug_df_with_disease['OncotreePrimaryDisease'] != disease].drop(columns=['ModelID', 'OncotreePrimaryDisease'])
+
+    # Perform t-tests comparing current disease vs the rest for each drug
+    t_test_results = ttest_ind(disease_drug_df, other_drug_df, axis=0, nan_policy='omit')
+    t_test_results_df = pd.DataFrame({
+        "drug": prism_df.columns,
+        "t_stat": t_test_results.statistic,
+        "p_value": t_test_results.pvalue
+    })
+
+    # Filter significant drugs based on p-value < 0.05
+    significant_drugs = t_test_results_df[t_test_results_df['p_value'] < 0.05]
+
+    # Perform ANOVA on these drugs
+    for drug in significant_drugs['drug'].unique():
+        # Extract drug responses for the current drug
+        disease_drug_responses = disease_drug_df[drug].dropna()
+        other_drug_responses = other_drug_df[drug].dropna()
+
+        # Perform ANOVA if both groups have sufficient data
+        if len(disease_drug_responses) > 1 and len(other_drug_responses) > 1:
+            f_statistic, p_value = f_oneway(disease_drug_responses, other_drug_responses)
+            higher_group = disease if disease_drug_responses.mean() > other_drug_responses.mean() else "Other Types"
+
+            # Store the results
+            ttest_results.append({
+                'OncotreePrimaryDisease': disease,
+                'Drug': drug,
+                'F-statistic': f_statistic,
+                'p-value': p_value,
+                'Higher in': higher_group
+            })
+
+# Convert results to DataFrame
+ttest_results_df = pd.DataFrame(ttest_results)
+
+# Apply a significance threshold (e.g., p < 0.05)
+significant_ttest_results_df = ttest_results_df[ttest_results_df['p-value'] < 0.05]
+
+# Display the top 50 significant results based on F-statistic
+significant_ttest_results_df.sort_values(by='F-statistic', key=abs, ascending=False).head(50)
+
+
+# In[7]:
+
+
+# Assuming 'drug_column_name' is the column in prism_trt_df that matches the 'drug' column in correlation_df
+prism_trt_df_filtered = prism_trt_df[['column_name', 'name', 'moa', 'target']]
+
+# Merge correlation_df with prism_trt_df based on the 'drug' column in correlation_df and the matching column in prism_trt_df
+merged_df = pd.merge(significant_ttest_results_df, prism_trt_df_filtered, how='left', left_on='Drug', right_on='column_name')
+
+# Drop the redundant drug_column_name column after the merge if needed
+merged_df = merged_df.drop(columns=['column_name'])
+
+# Save results to a CSV file
+merged_df.to_csv("../5.drug-dependency/results/drug_diff_results.csv", index=False)
+
+
+# In[8]:
+
+
+# Assuming merged_df and correlation_df are already defined
```

Comment:
Recommend updating this comment to be more specific to what this is doing (e.g., filtering data frame for specific columns).

Recommend confirming that all comments have this info too, it might be better for someone to understand the code.

Reply:
Done!